safe/gui/analysis_utilities.py

```diff
@@ -26,151 +32,95 @@
-        impact_function.impact.extent()
-
-    # generate report folder
-
-    # no other option for now
-    # TODO: retrieve the information from data store
-    if isinstance(impact_function.datastore.uri, QDir):
-        layer_dir = impact_function.datastore.uri.absolutePath()
-    else:
-        # No other way for now
-        return
+        disclaimer_text = setting('reportDisclaimer', None, str)
+        impact_report.inasafe_context.disclaimer = disclaimer_text
 
-    # We will generate it on the fly without storing it after datastore
-    # supports
-    impact_report.output_folder = os.path.join(layer_dir, 'output')
-    return impact_report.process_components()
+        north_arrow_path = setting('north_arrow_path', None, str)
+        impact_report.inasafe_context.north_arrow = north_arrow_path
 
+        # get the extent of impact layer
+        impact_report.qgis_composition_context.extent = \
+            impact_function.impact.extent()
 
-def generate_infographic_report(impact_function, iface):
-    """Generate impact map pdf from impact function.
+        # generate report folder
 
-    :param impact_function: The impact function used.
-    :type impact_function: ImpactFunction
+        # no other option for now
+        # TODO: retrieve the information from data store
+        if isinstance(impact_function.datastore.uri, QDir):
+            layer_dir = impact_function.datastore.uri.absolutePath()
+        else:
+            # No other way for now
+            return
 
-    :param iface: QGIS QGisAppInterface instance.
-    :type iface: QGisAppInterface
-    """
-    # get the extra layers that we need
-    extra_layers = []
-    print_atlas = setting('print_atlas_report', False, bool)
-    if print_atlas:
-        extra_layers.append(impact_function.aggregation_summary)
-    # create impact report instance
-    report_metadata = ReportMetadata(
-        metadata_dict=update_template_component(infographic_report))
-    impact_report = ImpactReport(
-        iface,
-        report_metadata,
-        impact_function=impact_function,
-        extra_layers=extra_layers)
-
-    # get the extent of impact layer
-    impact_report.qgis_composition_context.extent = \
-        impact_function.impact.extent()
-
-    # generate report folder
-
-    # no other option for now
-    # TODO: retrieve the information from data store
-    if isinstance(impact_function.datastore.uri, QDir):
-        layer_dir = impact_function.datastore.uri.absolutePath()
-    else:
-        # No other way for now
-        return
+        # We will generate it on the fly without storing it after datastore
+        # supports
+        impact_report.output_folder = os.path.join(layer_dir, 'output')
+        error_code, message = impact_report.process_components()
+        if error_code == ImpactReport.REPORT_GENERATION_FAILED:
+            break
+
+    remove_layer_from_canvas(map_overview_layer, impact_function)
```

Comment:
`map_overview_layer` will raise an exception if it's not a population analysis because it seems uninitialized!
Correct me if I'm wrong, it's difficult to read code on github